pkg/api/types.go

```diff
@@ -717,6 +721,113 @@ func (p *Properties) HasVMSSAgentPool() bool {
 	return false
 }
 
+// K8sOrchestratorName returns the 3 character orchestrator code for kubernetes-based clusters.
+func (p *Properties) K8sOrchestratorName() string {
+	if p.OrchestratorProfile.IsKubernetes() ||
+		p.OrchestratorProfile.IsOpenShift() {
+		if p.HostedMasterProfile != nil {
+			return "aks"
+		} else if p.OrchestratorProfile.IsOpenShift() {
+			return DefaultOpenshiftOrchestratorName
+		} else {
+			return DefaultOrchestratorName
+		}
+	}
+	return ""
+}
+
+func (p *Properties) getAgentVMPrefix() string {
+	return p.K8sOrchestratorName() + "-agentpool-"
+}
+
+func (p *Properties) getMasterVMPrefix() string {
+	return p.K8sOrchestratorName() + "-master-"
+}
+
+func (p *Properties) getVMPrefix() string {
+	if p.IsHostedMasterProfile() {
+		return p.getAgentVMPrefix()
+	}
+	return p.getMasterVMPrefix()
+}
+
+// GetRouteTableName returns the route table name of the cluster.
+func (p *Properties) GetRouteTableName() string {
+	return p.getVMPrefix() + "routetable"
+}
+
+// GetNSGName returns the name of the network security group of the cluster.
+func (p *Properties) GetNSGName() string {
+	return p.getVMPrefix() + "nsg"
+}
+
+// GetPrimaryAvailabilitySetName returns the name of the primary availability set of the cluster
+func (p *Properties) GetPrimaryAvailabilitySetName() string {
+	return p.AgentPoolProfiles[0].Name + "-availabilitySet-" + p.GenerateClusterID()
+}
+
+// GetPrimaryScaleSetName returns the name of the primary scale set node of the cluster
+func (p *Properties) GetPrimaryScaleSetName() string {
+	return p.K8sOrchestratorName() + "-" + p.AgentPoolProfiles[0].Name + "-" + p.GenerateClusterID() + "-vmss"
+}
+
+// IsHostedMasterProfile returns true if the cluster has a hosted master
+func (p *Properties) IsHostedMasterProfile() bool {
+	return p.HostedMasterProfile != nil
+}
+
+// GetSubnetName returns the subnet name of the cluster based on its current configuration.
+func (p *Properties) GetSubnetName() string {
```

Comment:
I don't see most of these functions being used in the var file yet, is it intentional?

Reply:
yes. This is intended to be non-breaking change. If everyone is okay with what we have so far. We can start using these methods to extract the output parameters in the templates.